corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -761,6 +762,8 @@ def extra_filters(self, for_count=False):
             filters.append(Q(server_modified_on__lt=self.end_date))
         if self.case_type is not None:
             filters.append(Q(type=self.case_type))
+        if self.case_ids:
+            filters.append(Q(case_id__in=self.case_ids))
```

Comment:
Rather than add this here you should use `CaseAccessor.iter_cases`

Reply:
I was not sure how to integrate that with rest of the structure. My understanding was that classes inherited from `ReindexAccessor` have a certain structure that is needed to use sql iterator?

Reply:
You could just create a new [DocumentProvider](https://github.com/dimagi/commcare-hq/blob/4a6f4cede4c01f482fe9dc9857e754c65f433325/corehq/util/doc_processor/interface.py#L61) class that takes a list of case_ids.

Reply:
https://github.com/dimagi/commcare-hq/pull/26610/commits/90d36e8b9e4004bf89f277a654db60eb2962cd4a